panic pulling artifact

While reviewing logs I saw a panic related to pulling:

```
panic: runtime error: index out of range [0] with length 0

goroutine 613994 [running]:
github.com/goharbor/harbor/src/controller/event/handler/internal.(*Handler).updatePullTime(0xc0000107f8, 0x20c6ca0, 0xc00120fb00, 0xc000447d50)
	/harbor/src/controller/event/handler/internal/artifact.go:70 +0x3cf
github.com/goharbor/harbor/src/controller/event/handler/internal.(*Handler).onPull.func1(0xc0000107f8, 0x20c6ca0, 0xc00120fb00, 0xc000447d50)
	/harbor/src/controller/event/handler/internal/artifact.go:53 +0x49
created by github.com/goharbor/harbor/src/controller/event/handler/internal.(*Handler).onPull
	/harbor/src/controller/event/handler/internal/artifact.go:53 +0x61
```

**Versions:**
Please specify the versions of following systems.

- harbor version: [2.0.1]

--- a/src/controller/event/handler/internal/artifact.go
+++ b/src/controller/event/handler/internal/artifact.go
@@ -67,7 +67,9 @@ func (a *Handler) updatePullTime(ctx context.Context, event *event.ArtifactEvent
 		if err != nil {
 			log.Infof("failed to list tags when to update pull time, %v", err)
 		} else {
-			tagID = tags[0].ID
+			if len(tags) != 0 {
+				tagID = tags[0].ID
+			}
 		}
 	}
 	if err := artifact.Ctl.UpdatePullTime(ctx, event.Artifact.ID, tagID, time.Now()); err != nil {

